ecosystem/sep-0030.md

```diff
@@ -467,9 +475,18 @@ available for signing. An account should check what signers are available
 periodically and update the signer on the account to be the most recently
 generated signer.
 
-The transaction must only contain operations for the account. The source
-account of the transaction and the source account of all operations in the
-transaction must match the account in the request.
+A server must only authorize operations in signed transactions that operate on
+the registered account that the client has authenticated for. This is to ensure
+that the server does not authorize operations for an account the client is not
+authorized. Generally to achieve this the transaction must only be signed if it
+contain operations for the authenticated account. The server can verify that
+the source account of the transaction and the source account of all operations
+in the transaction match the account in the request. In some cases servers may
+need to sign transactions that contain other accounts, like a sponsoring
+account used with CAP-33. In those cases a server may be implemented to allow
+transactions to contain that specific sponsoring account as a source account,
+but the server operator must ensure that the signing keys used by the server
+are never signers of the sponsoring account.
```

Comment:
I would split the generalized and concrete examples into two different sections. After re-reading this several times I'm starting to get it, but this paragraph leaves the implementor to interpret what other valid operations could have non-request source account.

Could we define exactly what operations could be permitted? Maybe thats too restrictive, but it feels like there are quasi-defined edge cases lingering here.

Reply:
I agree, I think that is too restrictive. It is also harder to implement and the additional requirements/limitations that would add don't actually help to prevent the problem and the approach isn't applicable to other non-sponsorship use cases that might have the same problem.

I've broken up the large paragraph and reworded some of it to try and be more direct. See 9540af0.

Reply:
This is much better, thanks.